engine/src/main/java/io/camunda/zeebe/engine/processing/common/ExpressionProcessor.java

```diff
@@ -335,6 +335,29 @@ public Either<Failure, DirectBuffer> evaluateVariableMappingExpression(
         .map(EvaluationResult::toBuffer);
   }
 
+  /**
+   * Evaluates the given expression of a script task and returns the result as buffer. If the
+   * evaluation fails or the result is not a context then a failure is returned.
+   *
+   * @param expression the expression to evaluate
+   * @param scopeKey the scope to load the variables from (a negative key is intended to imply an
+   *     empty variable context)
+   * @return either the evaluation result as buffer, or a failure
+   */
+  public Either<Failure, DirectBuffer> evaluateScriptExpression(
+      final Expression expression, final long scopeKey) {
+    final var expectedTypes =
+        List.of(
+            ResultType.BOOLEAN,
+            ResultType.NUMBER,
+            ResultType.STRING,
+            ResultType.ARRAY,
+            ResultType.OBJECT);
```

Comment:
:x: We should allow all result types. 

No need to limit the expression result to `boolean`, `number`, `string`, `array`, or `object`. Also `null` and temporal values should be possible.

Reply:
Hi @saig0. Please check this out. 🙇